update doku (#4)
930C/PySmartHome: docs/doku.md

            alt value below constant
                FertilizationController->>Fertilizer: fertilize(value: float)            
            else value above constant and device running
                FertilizationController->>Fertilizer: stop_fertilizing()
            end
        FertilizationController->>FertilizationSensor: update()
    end
```
Das Sequenzdiagramm beschreibt beispielhaft den Prozess, wie sich der `FertilizationController` in unserem "smarten Gewächshaus" verhält, um die Düngung (Fertilization) steuern:
- Der Prozess beginnt, wenn die Main-Funktion den `SmartHomeController` aufruft, um die Methode `update()` auszuführen. Der `SmartHomeController` ist verantwortlich für die Gesamtsteuerung und Verwaltung des Smart Home Systems.

- In einem wiederholten Schleifenablauf, der alle 5 Sekunden stattfindet, ruft der `SmartHomeController` den `FertilizationController` auf, um die Düngungssteuerung durchzuführen.
Das Sequenzdiagramm beschreibt den Prozess, wie der "SmartHomeController" und der "FertilizationController" in regelmäßigen Abständen miteinander interagieren, um die Düngung (Fertilization) einer Pflanze zu steuern. Hier ist eine Beschreibung des Prozesses:

> - Der `FertilizationController` beginnt, indem er den `FertilizationSensor` aufruft, um den aktuellen Düngungswert zu berechnen. Dieser berechnet den Wert und gibt ihn an den `FertilizationController` zurück.

Reviewer: ```suggestion
- Der `FertilizationController` beginnt, indem er alle Sensorwerte aller `FertilizationSensor` abruft. Der Sensor ermittelt seinen eigenen Messwert und je nach Strategie wird der Mittel-, Maximal- oder Minimalwert aller Sensoren in der Zone berechnet und an den `FertilizationController` zurückgegeben.
```

- Nachdem der Düngungswert ermittelt wurde, wird eine Bedingung geprüft, ob der aktuelle Wert unter einem vorgegebenen Schwellenwert liegt oder nicht.

>    - Wenn der Wert unter dem Schwellenwert liegt, ist es notwendig, zu düngen. Der `FertilizationController` ruft die Methode `fertilize(value: float)` des "Fertilizer" auf, um die Düngung mit dem berechneten Wert durchzuführen. Der `Fertilizer` ist das Gerät oder die Komponente, die für die tatsächliche Düngung der Pflanze verantwortlich ist.

Reviewer: ```suggestion
   - Wenn der Wert unter dem Schwellenwert liegt, ist es notwendig, zu düngen. Der `FertilizationController` ruft die Methode `fertilize(value: float)` des `Fertilizer` auf, um die Düngung mit dem berechneten Wert durchzuführen. Der `Fertilizer` ist das Gerät oder die Komponente, die für die tatsächliche Düngung der Pflanze verantwortlich ist.
```

   - Wenn der Wert über dem Schwellenwert liegt und das Düngungsgerät bereits läuft, dann muss die Düngung gestoppt werden. Der `FertilizationController` ruft die Methode `stop_fertilizing()` des `Fertilizer` auf, um die Düngung zu beenden.

> - Der Schleifenablauf wird alle 5 Sekunden wiederholt, wodurch die Düngung der Pflanze periodisch überwacht und gesteuert wird.

Reviewer: ```suggestion
- Der Schleifenablauf wird alle 5 Sekunden wiederholt, wodurch die Düngung der Pflanzen periodisch überwacht und gesteuert wird.
```

# 7 Verteilungssicht {#section-deployment-view}
Aufgrund der Anforderungen an das Projekt wurden keine  Test, Development oder Produktivumgebung aufgesetzt.
Das Smart Home System ist zum jetztigen Stand ein lokal funktionierendes System, dass in Python geschrieben wurde und die Abläufe simuliert.
Wenn das System produktiv eingesetzt werden würde, müsste man die Sensoren sowie die smarten Geräte als Akteure eines verteilten Systems betrachten, die Daten sammeln. In diesem Zusammenhang ist es erfolderlich diese Daten in einer Server Struktur zu konsolidieren, die das lauffähige System bereitstellt. Das könnte in dieser Form umzusetzen sein.
```mermaid
flowchart TD
    subgraph Smart Home 
        A[Smart Home System]<-->Heater
        B[Temperature Sensor]-->A[Smart Home System]
        A[Smart Home System]<-->Fan
        A[Smart Home System]<-->C[Roller Blind]
        subgraph ide1 [ServerCluster]
            A[Smart Home System]-->id1[(Database)]
        end
    end
```

> Da unser System die Veränderungern der Sensoren lediglich simuliert und nicht mit realen Events arbeitet, würde das aktuelle System nicht ohne Anpassungen in dieser Form darstellbar sein. Jedoch schafft diese Visualisierung ein gutes Bild, wohin die Architektur gehen kann 

Reviewer: ```suggestion
Da unser System die Veränderungern der Sensoren lediglich simuliert und nicht mit realen Events arbeitet, würde das aktuelle System nicht ohne Anpassungen in dieser Form darstellbar sein. Jedoch schafft diese Visualisierung ein gutes Bild, wohin die Architektur gehen könnte.
```
